UnityProject/Assets/Scripts/Objects/Machines/MaterialStorage.cs

```diff
@@ -36,8 +40,9 @@ private void AddMaterial(ItemTrait material, int quantity)
 
 		private void ConsumeMaterial(ItemTrait material, int quantity)
 		{
-			MaterialList[material] -= quantity;
-			currentResources -= quantity;
+			var quantityToSubtract = (int) Mathf.Clamp(quantity, 0, Single.PositiveInfinity);
```

Comment:
surely clamping to how much material you have would be better way of doing it?
so if you got 50 clamps maximum to 50?

Reply:
It's a safety precaution to make sure quantity never falls below 0. It's basically doing the same exact thing as before but making sure that quantity is never negative which the machine will try to do sometimes.

Reply:
how does it become negative?

Reply:
20 materials - 50 materials to consume = -30 quantity

Reply:
meant as in quantity , and surely you could just do 99999
 and then that would still equal
 MaterialList[material] = -99994

Reply:
I reverted it because I can't replicate the issue anymore.